pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -118,6 +121,32 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 		return err
 	}
 
+	tmpdir, err := os.MkdirTemp("", pxeAssetsPath)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	dstfilename := filepath.Join(tmpdir, "config.cfg")
+	err = a.extractPXEFileFromISO(a.isoPath, "EFI/redhat/grub.cfg", dstfilename)
```

Comment:
What about (logically) post-poning this PR after https://github.com/openshift/installer/pull/7185, so that it could be possible to read directly the artifacts already extracted in the temp dir

Reply:
Another point: wouldn't be recommended to read `/coreos/kargs.json` for extracting the kernel parameters? It should be also simpler to parse, being a json file, ie something like: 

```
{
  "default": "coreos.liveiso=rhcos-414.92.202305090606-0 ignition.firstboot ignition.platform.id=metal",
  "files": [
    {
      "offset": 968,
      "path": "EFI/redhat/grub.cfg"
    },
    {
      "offset": 1865,
      "path": "isolinux/isolinux.cfg"
    }
  ],
  "size": 1112
}
```

Reply:
Done